### PR TITLE
Fixes in Conformance chapter

### LINKED ISSUE #1045: Update Conformance chapter
Some things to fix in the [Conformance](https://github.com/spdx/spdx-spec/blob/development/v3.0.1/docs/conformance.md) chapter.

- remove (or at least check) the numbering of profiles chapters in second bullet list
- in the same section, fix that only Core is mandatory
- add the section on trademark conformance, from SPDXv2 standard
- In Lite profile, mention that Software also needs to be compliant
- delete the last section, since this is about serializations -- and we definitely do not want to mention Tag-Value or Spreadsheets as "current supported formats"

plus a couple of other minor points.

### PR DESCRIPTION
**Pull request by zvr**
- adds the section on trademark conformance, from SPDXv2 standard
- remove the numbering of profiles chapters in second bullet list
- in the same section, fixes that only Core is mandatory
- delete the last section, since this is about serializations -- and we definitely do not want to mention Tag-Value or Spreadsheets as "current supported formats"

Fixes #1045

Signed-off-by: Alexios Zavras <github@zvr.gr>

### CODE REVIEW

**Comment by zvr** [comment]
Depending on the order of merging (the number removal and the link to json), this might have conflicts, but they should be obvious.

File: docs/conformance.md [hunk 1]
```diff
@@ -200,100 +207,24 @@ of the SPDX.
...
-Conformance to the Extension Profile compliance point does not entail support
-for the Licencing, Security, Dataset, AI, Build, or profiles of the SPDX but
-is expected to be used in combination with the other profiles to extend them.
-
-This compliance point facilitates interchange of extended information that goes
-beyond the standard SPDX produced by tools supporting SPDX and is used between
-cooperating parties that understand the form of the extension and can produce
-and consume its non-standard content.
-
-## 5.12 Standard data format requirements <a name="5.12"></a>
```
Comment (kestewart) [inline docs/conformance.md:246]: I think we should retain this section.   Why did you choose to delete it?

File: docs/conformance.md [hunk 2]
```diff
@@ -200,100 +207,24 @@ of the SPDX.
 This compliance point facilitates interchange of minimal licencing information
 when produced by tools supporting SPDX.
 
-## 5.11 Extension Profile compliance point <a name="5.11"></a>
```
Comment (kestewart) [inline docs/conformance.md:203]: When was it determined to remove this section?

**Review by kestewart — CHANGES_REQUESTED** [review]
All of the additions make sense and am good with, however not sure why Lite & Extension Profile Conformance point information was removed?

File: docs/conformance.md (hunk 2 above)
Comment (zvr) [reply docs/conformance.md:203]: Ooops, that was not the intent.

Comment (bact) [reply docs/conformance.md]: In this section, we still have to figure out the standard naming convention of the SPDX files (Lines 286-295) -- if multiple extensions (`.spdx.json`) are ok for OMG. 

See issue: https://github.com/spdx/spdx-spec/issues/987 We can do it in a seperate PR.

**Review by goneall — APPROVED** [review]
LGTM

**Review by kestewart — APPROVED** [review]
Per discussions, Standard Data Format will be added back in future version, after we sort through the serialization chapter.

**Review by bobmartin3000 — APPROVED** [review]
LGTM

**Review by bact — COMMENTED** [review]
LGTM - the information about data format is covered enough for now in Serialization chapter

**Review by bact — APPROVED** [review]

**Pull request merged**